File: codecamp/src/main/java/com/gdogaru/codecamp/model/Schedule.java
package com.gdogaru.codecamp.model;

import org.joda.time.LocalDateTime;

import java.util.List;

/**
 * Created by Gabriel on 10/2/2016.
 */

public class Schedule {
    LocalDateTime date;
    List<TimeSlot> timeSlots;
    List<Track> tracks;
    List<Session> sessions;

    public LocalDateTime getDate() {
        return date;
    }

    public void setDate(LocalDateTime date) {
        this.date = date;
    }

    public List<TimeSlot> getTimeSlots() {
        return timeSlots;
    }

    public void setTimeSlots(List<TimeSlot> timeSlots) {
        this.timeSlots = timeSlots;
    }

    public List<Track> getTracks() {
        return tracks;
    }

    public void setTracks(List<Track> tracks) {
        this.tracks = tracks;
    }

    public List<Session> getSessions() {
        return sessions;
    }

    public void setSessions(List<Session> sessions) {
        this.sessions = sessions;
    }
}

File: codecamp/src/main/java/com/gdogaru/codecamp/api/model/Schedule.kt
package com.gdogaru.codecamp.api.model


import com.fasterxml.jackson.annotation.JsonProperty
import org.threeten.bp.LocalDateTime
import org.threeten.bp.LocalTime

/**
 * Created by Gabriel on 10/2/2016.
 */

data class Schedule(
        var date: LocalDateTime,
        var timeSlots: List<TimeSlot>,
        var tracks: List<Track>,
        var sessions: List<Session>
)


data class TimeSlot(
        var startTime: LocalTime,
        var endTime: LocalTime
)

data class Track(
        @JsonProperty("name")
        var name: String,

        @JsonProperty("capacity")
        var capacity: String,

        @JsonProperty("description")
        var description: String,

        @JsonProperty("displayOrder")
        var displayOrder: Int = 0
)


data class Session(
        @JsonProperty("title")
        var title: String,

        @JsonProperty("description")
        var description: String?,

        @JsonProperty("startTime")
        var startTime: LocalTime?,

        @JsonProperty("endTime")

        var endTime: LocalTime?,
        @JsonProperty("speakers")
        var speakerIds: List<String>?,

        @JsonProperty("speakingLang")
        var speakingLang: String?,

        @JsonProperty("level")
        var level: String?,

        @JsonProperty("allTracks")
        var allTracks: Boolean? = null,

        @JsonProperty("track")
        var track: String?

) {
    val id: String
        get() = startTime.toString() + track

    companion object {

        val SESSION_BY_DATE_COMPARATOR: java.util.Comparator<Session> = Comparator { object1, object2 ->
            object1.startTime?.compareTo(object2.startTime) ?: -1
        }
    }
}
